app/controllers/api/v2/compliance/arf_reports_controller.rb

```diff
@@ -84,12 +80,27 @@ def find_resource
           instance_variable_set("@arf_report", resource_scope.find(params[:id]))
         end
 
+        def find_resources_before_create
+          @asset = ForemanOpenscap::Helper::get_asset(params[:cname], params[:policy_id])
+          if !params[:openscap_proxy_url] && !params[:openscap_proxy_name]
+            msg = _('Failed to upload Arf Report, OpenSCAP proxy name or url not found in params for when uploading report for %s') % @asset.host.name
+            no_proxy_for_host(msg)
+            return
+          end
+          @smart_proxy = SmartProxy.find_by :name => params[:openscap_proxy_name]
```

Comment:
how do we set current org/loc here? should we use `.unscoped` to always find a proxy, context regardless?